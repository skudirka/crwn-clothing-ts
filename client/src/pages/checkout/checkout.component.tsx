import React from 'react';
import {connect, ConnectedProps} from 'react-redux';
import {createStructuredSelector} from 'reselect';

import CheckoutItem from '../../components/checkout-item/checkout-item.component';
import StripeCheckoutButton from '../../components/stripe-button/stripe-button.component';

import {selectCartItems, selectCartTotal} from '../../redux/cart/cart.selectors';
import {IRootState} from '../../redux/root-types';
import {ICartItem} from '../../redux/cart/cart.types';

import {
    CheckoutPageContainer,
    CheckoutHeaderContainer,
    HeaderBlockContainer,
    TotalContainer,
    WarningContainer
} from './checkout.styles';

interface mapStateToPropsInterface {
    cartItems:ICartItem[];
    total:number;
}
const mapStateToProps = createStructuredSelector<IRootState, mapStateToPropsInterface>({
    cartItems: selectCartItems,
    total: selectCartTotal
});

const connector = connect(
    mapStateToProps
);

type PropsFromRedux = ConnectedProps<typeof connector>;

type CheckoutPageProps = PropsFromRedux & {};

export const CheckoutPage:React.FC<CheckoutPageProps> = ({cartItems, total}) => (
    <CheckoutPageContainer className="checkout-page">
        <CheckoutHeaderContainer className="checkout-header">
            <HeaderBlockContainer className="header-block">
                <span>Product</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer className="header-block">
                <span>Description</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer className="header-block">
                <span>Quantity</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer className="header-block">
                <span>Price</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer className="header-block">
                <span>Remove</span>
            </HeaderBlockContainer>
        </CheckoutHeaderContainer>
        {
            cartItems.map(cartItem => (
                <CheckoutItem key={cartItem.id} cartItem={cartItem} />
            ))
        }

        <TotalContainer className="total">TOTAL: ${total}</TotalContainer>
        <WarningContainer className="test-warning">
            *Please use the following test credit card for payments* <br />
            4242 4242 4242 4242 - Exp: 01/20 - CVV: 123
        </WarningContainer>
        <StripeCheckoutButton price={total} />
    </CheckoutPageContainer>
);



export default connector(CheckoutPage);