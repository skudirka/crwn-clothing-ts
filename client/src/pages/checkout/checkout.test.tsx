import {shallow, mount} from 'enzyme';
import React from 'react';

import {CheckoutPage} from './checkout.component';

describe('CheckoutPage page', () => {
    let wrapper;
    let mockDispatch;
    let mockProps;

    beforeEach(() => {
        mockDispatch = jest.fn();

        mockProps = {
            cartItems: [{ id: 1 }, { id: 2 }, { id: 3 }],
            total: 3
        };

        wrapper = shallow(<CheckoutPage {...mockProps} />);
    });

    it('should render CheckoutPage page', () => {
        expect(wrapper).toMatchSnapshot();
    });
})