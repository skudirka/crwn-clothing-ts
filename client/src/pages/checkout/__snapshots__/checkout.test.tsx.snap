// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPage page should render CheckoutPage page 1`] = `ShallowWrapper {}`;
